File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        
        while(true){
            if(answer > 500) return -1;
            
            if (num == 1) break;
            else if(num % 2 == 1) num = num * 3 + 1;
            else num /= 2;
            answer++;
        }
        
        return answer;
    }
}

File: Python3/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    cnt = -1
    
    for i in range(0, 500):
        if num == 1:
            cnt = i
            break
        
        if num % 2 == 0:
            num /= 2
        else:
            num = num * 3 + 1
    
    return cnt